# Installing and Importing necessary Packages


In [ ]:
!pip install -U scikit-learn


Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import accuracy_score, confusion_matrix
from sklearn import metrics
from random import randint

# removing the lines below will introduce gaussian naive beyes input feature warning
# but everything still works fine
import warnings
warnings.filterwarnings("ignore", category=UserWarning)


#**Diabetes Dataset**
this set of data is composed of the glucose amount and bloodpressure to know if the patient has a diabetes.

#Reading Dataset

In [ ]:
df = pd.read_csv('/content/fexp1/Naive-Bayes-Classification-Data.csv')
df

Glucose  BloodPressure  Diabetes
0        148             72         1
1         85             66         0
2        183             64         1
3         89             66         0
4        137             40         1
..       ...            ...       ...
990      113             80         0
991      138             82         0
992      108             68         0
993       99             70         0
994      103             72         0

[995 rows x 3 columns]

#Training the Model


In [ ]:
x = df.drop('Diabetes', axis = 1)
y = df['Diabetes']

In [ ]:
from sklearn.model_selection import train_test_split

In [ ]:
x_train, x_test, y_train, y_test =train_test_split(x,y,test_size=0.2)

In [ ]:
x_train

Glucose  BloodPressure
567       92             62
989      148             60
614      138             74
224      100             66
336      117              0
..       ...            ...
264      123             62
414      138             60
77        95             72
530      122             60
285      136             74

[796 rows x 2 columns]

In [ ]:
x_test

Glucose  BloodPressure
326      122             64
440      189            104
135      125             60
428      135             94
151      114             65
..       ...            ...
22       196             90
900      111             62
276      106             60
825      158             84
240       91             64

[199 rows x 2 columns]

In [ ]:
y_train

567    0
989    1
614    1
224    0
336    0
      ..
264    1
414    1
77     0
530    0
285    0
Name: Diabetes, Length: 796, dtype: int64

In [ ]:
y_test

326    1
440    1
135    0
428    0
151    0
      ..
22     1
900    0
276    1
825    1
240    0
Name: Diabetes, Length: 199, dtype: int64

In [ ]:
model = GaussianNB()
model.fit(x_train, y_train)

GaussianNB()

In [ ]:
predicted_value = model.predict(x_test)
predicted_value

array([0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 1, 0, 1, 0, 0,
       0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0,
       0, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 1, 0, 0, 0,
       0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1,
       0, 1, 1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 1, 1,
       0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 1, 0, 0, 1, 0, 1, 0, 0, 0,
       0, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 1,
       0])

#Accuracy Score


In [ ]:
accuracy_score(predicted_value,y_test)*100

77.38693467336684

In [ ]:
#Alternative code
tp, fp, tn, fn = confusion_matrix(y_test, predicted_value).ravel()
print(int((tp+tn)/199*100),"%")

75 %


#Confusion Matrix

In [ ]:
from sklearn.metrics import confusion_matrix, accuracy_score

In [ ]:
confusion_matrix(predicted_value,y_test)

array([[116,  35],
       [ 10,  38]])

#Model Testing

In [ ]:
df_sample = pd.DataFrame(columns=['Glucose', 'BloodPressure'])
glucose = input('Enter Glucose:')
bloodpressure = input('Enter Blood Pressure:')
df_sample.loc[len(df_sample.index)] = [glucose, bloodpressure]

Enter Glucose:148
Enter Blood Pressure:72


In [ ]:
df_sample

Glucose BloodPressure
0     148            72

#Model Prediction

In [ ]:
model.predict(df_sample)

array([1])

#Sample Python Program

In [ ]:
prediction=model.predict(df_sample)
if prediction == 0:
  print("Not Diabetic")
else:
  print("Diabetic")

Diabetic


#Recall

In [ ]:
cf = confusion_matrix(y_test, predicted_value)
cf[0,0]/(cf[0,0] + cf[1,0])*100

76.82119205298014

In [ ]:
#Alternative code
tp, fp, tn, fn = confusion_matrix(y_test, predicted_value).ravel()
print(int(tp/(tp+fn)*100),"%")

75 %


#Prescision

In [ ]:
cf = confusion_matrix(predicted_value,y_test)
cf[0,0]/(cf[0,0] + cf[1,1])*100

75.32467532467533

In [ ]:
#Alternative code
tp, fp, tn, fn = confusion_matrix(y_test, predicted_value).ravel()
print(int(tp/(tp+fp)*100),"%")

92 %


#Computation and result of Type 1 and 2 errors

In [ ]:
#case 1
df_sample = pd.DataFrame(columns=['Glucose', 'BloodPressure'])
glucose = input('Enter Glucose:')
bloodpressure = input('Enter Blood Pressure:')
df_sample.loc[len(df_sample.index)] = [glucose, bloodpressure]
prediction=model.predict(df_sample)
if prediction == 0:
  print("Congratulations, You are not Diabetic.\n")
  print("Recommendations for preventing diabetes:")
  print("- Maintain a healthy weight through regular exercise and a balanced diet, as excess weight can increase the risk of developing diabetes.")
  print("- Eat a balanced diet that is rich in fruits, vegetables, whole grains, lean protein, and healthy fats, and limit processed and sugary foods.")
  print("- Exercise regularly, as it can help improve insulin sensitivity, lower blood sugar levels, and improve overall health.")
  print("- Quit smoking, as smoking can increase the risk of developing diabetes.")
  print("- Monitor blood pressure and cholesterol levels, as both can contribute to the development of diabetes and other health problems.")
  print("- Manage stress levels, as stress can impact overall health.")

else:
  print("I'm sorry to tell that you are Diabetic.\n")
  print("Recommendations for managing diabetes and promoting overall health:")
  print("- Monitor blood sugar levels through monitoring, medication, and lifestyle changes such as regular exercise and a healthy diet.")
  print("- Eat a balanced diet that is rich in fruits, vegetables, whole grains, lean protein, and healthy fats, and limit processed and sugary foods.")
  print("- Exercise regularly, as it can help improve insulin sensitivity, lower blood sugar levels, and improve overall health.")
  print("- Quit smoking, as smoking can increase the risk of complications associated with diabetes, such as heart disease and nerve damage.")
  print("- Monitor blood pressure and cholesterol levels, as both can contribute to the development of complications associated with diabetes.")
  print("- Manage stress levels, as stress can impact blood sugar levels and overall health.")


Enter Glucose:148
Enter Blood Pressure:72
I'm sorry to tell that you are Diabetic.

Recommendations for managing diabetes and promoting overall health:
- Monitor blood sugar levels through monitoring, medication, and lifestyle changes such as regular exercise and a healthy diet.
- Eat a balanced diet that is rich in fruits, vegetables, whole grains, lean protein, and healthy fats, and limit processed and sugary foods.
- Exercise regularly, as it can help improve insulin sensitivity, lower blood sugar levels, and improve overall health.
- Quit smoking, as smoking can increase the risk of complications associated with diabetes, such as heart disease and nerve damage.
- Monitor blood pressure and cholesterol levels, as both can contribute to the development of complications associated with diabetes.
- Manage stress levels, as stress can impact blood sugar levels and overall health.


> This is a case of True Positive

Because the Diabetes Value is 1 and the given variables for blood pressure(72) as well as glucose(148) are correct

In [ ]:
#case 2
df_sample = pd.DataFrame(columns=['Glucose', 'BloodPressure'])
glucose = input('Enter Glucose:')
bloodpressure = input('Enter Blood Pressure:')
df_sample.loc[len(df_sample.index)] = [glucose, bloodpressure]
prediction=model.predict(df_sample)
if prediction == 0:
  print("Congratulations, You are not Diabetic.\n")
  print("Recommendations for preventing diabetes:")
  print("- Maintain a healthy weight through regular exercise and a balanced diet, as excess weight can increase the risk of developing diabetes.")
  print("- Eat a balanced diet that is rich in fruits, vegetables, whole grains, lean protein, and healthy fats, and limit processed and sugary foods.")
  print("- Exercise regularly, as it can help improve insulin sensitivity, lower blood sugar levels, and improve overall health.")
  print("- Quit smoking, as smoking can increase the risk of developing diabetes.")
  print("- Monitor blood pressure and cholesterol levels, as both can contribute to the development of diabetes and other health problems.")
  print("- Manage stress levels, as stress can impact overall health.")

else:
  print("I'm sorry to tell that you are Diabetic.\n")
  print("Recommendations for managing diabetes and promoting overall health:")
  print("- Monitor blood sugar levels through monitoring, medication, and lifestyle changes such as regular exercise and a healthy diet.")
  print("- Eat a balanced diet that is rich in fruits, vegetables, whole grains, lean protein, and healthy fats, and limit processed and sugary foods.")
  print("- Exercise regularly, as it can help improve insulin sensitivity, lower blood sugar levels, and improve overall health.")
  print("- Quit smoking, as smoking can increase the risk of complications associated with diabetes, such as heart disease and nerve damage.")
  print("- Monitor blood pressure and cholesterol levels, as both can contribute to the development of complications associated with diabetes.")
  print("- Manage stress levels, as stress can impact blood sugar levels and overall health.")


Enter Glucose:85
Enter Blood Pressure:66
Congratulations, You are not Diabetic.

Recommendations for preventing diabetes:
- Maintain a healthy weight through regular exercise and a balanced diet, as excess weight can increase the risk of developing diabetes.
- Eat a balanced diet that is rich in fruits, vegetables, whole grains, lean protein, and healthy fats, and limit processed and sugary foods.
- Exercise regularly, as it can help improve insulin sensitivity, lower blood sugar levels, and improve overall health.
- Quit smoking, as smoking can increase the risk of developing diabetes.
- Monitor blood pressure and cholesterol levels, as both can contribute to the development of diabetes and other health problems.
- Manage stress levels, as stress can impact overall health.


> This is a case of True Negative

Because the Diabetes Value is 0 and the given variables for blood pressure(66) as well as glucose(85) are correct

In [ ]:
#case 3
df_sample = pd.DataFrame(columns=['Glucose', 'BloodPressure'])
glucose = input('Enter Glucose:')
bloodpressure = input('Enter Blood Pressure:')
df_sample.loc[len(df_sample.index)] = [glucose, bloodpressure]
prediction=model.predict(df_sample)
if prediction == 0:
  print("Congratulations, You are not Diabetic.\n")
  print("Recommendations for preventing diabetes:")
  print("- Maintain a healthy weight through regular exercise and a balanced diet, as excess weight can increase the risk of developing diabetes.")
  print("- Eat a balanced diet that is rich in fruits, vegetables, whole grains, lean protein, and healthy fats, and limit processed and sugary foods.")
  print("- Exercise regularly, as it can help improve insulin sensitivity, lower blood sugar levels, and improve overall health.")
  print("- Quit smoking, as smoking can increase the risk of developing diabetes.")
  print("- Monitor blood pressure and cholesterol levels, as both can contribute to the development of diabetes and other health problems.")
  print("- Manage stress levels, as stress can impact overall health.")

else:
  print("I'm sorry to tell that you are Diabetic.\n")
  print("Recommendations for managing diabetes and promoting overall health:")
  print("- Monitor blood sugar levels through monitoring, medication, and lifestyle changes such as regular exercise and a healthy diet.")
  print("- Eat a balanced diet that is rich in fruits, vegetables, whole grains, lean protein, and healthy fats, and limit processed and sugary foods.")
  print("- Exercise regularly, as it can help improve insulin sensitivity, lower blood sugar levels, and improve overall health.")
  print("- Quit smoking, as smoking can increase the risk of complications associated with diabetes, such as heart disease and nerve damage.")
  print("- Monitor blood pressure and cholesterol levels, as both can contribute to the development of complications associated with diabetes.")
  print("- Manage stress levels, as stress can impact blood sugar levels and overall health.")


Enter Glucose:116
Enter Blood Pressure:74
Congratulations, You are not Diabetic.

Recommendations for preventing diabetes:
- Maintain a healthy weight through regular exercise and a balanced diet, as excess weight can increase the risk of developing diabetes.
- Eat a balanced diet that is rich in fruits, vegetables, whole grains, lean protein, and healthy fats, and limit processed and sugary foods.
- Exercise regularly, as it can help improve insulin sensitivity, lower blood sugar levels, and improve overall health.
- Quit smoking, as smoking can increase the risk of developing diabetes.
- Monitor blood pressure and cholesterol levels, as both can contribute to the development of diabetes and other health problems.
- Manage stress levels, as stress can impact overall health.


> This is a case of True Negative

Because the Diabetes Value is 1 and the given variables for blood pressure(72) as well as glucose(148) are correct

In [ ]:
#case 4
df_sample = pd.DataFrame(columns=['Glucose', 'BloodPressure'])
glucose = input('Enter Glucose:')
bloodpressure = input('Enter Blood Pressure:')
df_sample.loc[len(df_sample.index)] = [glucose, bloodpressure]
prediction=model.predict(df_sample)
if prediction == 0:
  print("Congratulations, You are not Diabetic.\n")
  print("Recommendations for preventing diabetes:")
  print("- Maintain a healthy weight through regular exercise and a balanced diet, as excess weight can increase the risk of developing diabetes.")
  print("- Eat a balanced diet that is rich in fruits, vegetables, whole grains, lean protein, and healthy fats, and limit processed and sugary foods.")
  print("- Exercise regularly, as it can help improve insulin sensitivity, lower blood sugar levels, and improve overall health.")
  print("- Quit smoking, as smoking can increase the risk of developing diabetes.")
  print("- Monitor blood pressure and cholesterol levels, as both can contribute to the development of diabetes and other health problems.")
  print("- Manage stress levels, as stress can impact overall health.")

else:
  print("I'm sorry to tell that you are Diabetic.\n")
  print("Recommendations for managing diabetes and promoting overall health:")
  print("- Monitor blood sugar levels through monitoring, medication, and lifestyle changes such as regular exercise and a healthy diet.")
  print("- Eat a balanced diet that is rich in fruits, vegetables, whole grains, lean protein, and healthy fats, and limit processed and sugary foods.")
  print("- Exercise regularly, as it can help improve insulin sensitivity, lower blood sugar levels, and improve overall health.")
  print("- Quit smoking, as smoking can increase the risk of complications associated with diabetes, such as heart disease and nerve damage.")
  print("- Monitor blood pressure and cholesterol levels, as both can contribute to the development of complications associated with diabetes.")
  print("- Manage stress levels, as stress can impact blood sugar levels and overall health.")


Enter Glucose:122
Enter Blood Pressure:90
Congratulations, You are not Diabetic.

Recommendations for preventing diabetes:
- Maintain a healthy weight through regular exercise and a balanced diet, as excess weight can increase the risk of developing diabetes.
- Eat a balanced diet that is rich in fruits, vegetables, whole grains, lean protein, and healthy fats, and limit processed and sugary foods.
- Exercise regularly, as it can help improve insulin sensitivity, lower blood sugar levels, and improve overall health.
- Quit smoking, as smoking can increase the risk of developing diabetes.
- Monitor blood pressure and cholesterol levels, as both can contribute to the development of diabetes and other health problems.
- Manage stress levels, as stress can impact overall health.


> This is a case of False Negative (Type 2 Error)

Because the Glucose Level of 122 and Blood Pressure of 90 is Diabetic in the Dataset.

In [ ]:
#case 5
df_sample = pd.DataFrame(columns=['Glucose', 'BloodPressure'])
glucose = input('Enter Glucose:')
bloodpressure = input('Enter Blood Pressure:')
df_sample.loc[len(df_sample.index)] = [glucose, bloodpressure]
prediction=model.predict(df_sample)
if prediction == 0:
  print("Congratulations, You are not Diabetic.\n")
  print("Recommendations for preventing diabetes:")
  print("- Maintain a healthy weight through regular exercise and a balanced diet, as excess weight can increase the risk of developing diabetes.")
  print("- Eat a balanced diet that is rich in fruits, vegetables, whole grains, lean protein, and healthy fats, and limit processed and sugary foods.")
  print("- Exercise regularly, as it can help improve insulin sensitivity, lower blood sugar levels, and improve overall health.")
  print("- Quit smoking, as smoking can increase the risk of developing diabetes.")
  print("- Monitor blood pressure and cholesterol levels, as both can contribute to the development of diabetes and other health problems.")
  print("- Manage stress levels, as stress can impact overall health.")

else:
  print("I'm sorry to tell that you are Diabetic.\n")
  print("Recommendations for managing diabetes and promoting overall health:")
  print("- Monitor blood sugar levels through monitoring, medication, and lifestyle changes such as regular exercise and a healthy diet.")
  print("- Eat a balanced diet that is rich in fruits, vegetables, whole grains, lean protein, and healthy fats, and limit processed and sugary foods.")
  print("- Exercise regularly, as it can help improve insulin sensitivity, lower blood sugar levels, and improve overall health.")
  print("- Quit smoking, as smoking can increase the risk of complications associated with diabetes, such as heart disease and nerve damage.")
  print("- Monitor blood pressure and cholesterol levels, as both can contribute to the development of complications associated with diabetes.")
  print("- Manage stress levels, as stress can impact blood sugar levels and overall health.")


Enter Glucose:125
Enter Blood Pressure:96
Congratulations, You are not Diabetic.

Recommendations for preventing diabetes:
- Maintain a healthy weight through regular exercise and a balanced diet, as excess weight can increase the risk of developing diabetes.
- Eat a balanced diet that is rich in fruits, vegetables, whole grains, lean protein, and healthy fats, and limit processed and sugary foods.
- Exercise regularly, as it can help improve insulin sensitivity, lower blood sugar levels, and improve overall health.
- Quit smoking, as smoking can increase the risk of developing diabetes.
- Monitor blood pressure and cholesterol levels, as both can contribute to the development of diabetes and other health problems.
- Manage stress levels, as stress can impact overall health.


> This is a case of False Negative (Type 2 Error)

Because the Glucose Level of 125 and Blood Pressure of 96 is Diabetic in the Dataset.